astropy/io/misc/hdf5.py

```diff
@@ -259,7 +260,10 @@ def write_table_hdf5(table, output, path=None, compression=False,
         raise Exception("h5py is required to read and write HDF5 files")
 
     if path is None:
-        raise ValueError("table path should be set via the path= argument")
+        # __astropy__ is just an arbitrary, hardcoded string here.
+        path = '__astropy__'
+        warnings.warn("table path was not set via the path= argument; "
+                      "using default path {}".format(path))
```

Comment:
I think it might make sense to check that the current file/group is empty, and only choose a default path in that case, but still emit an error otherwise. I think we should only pick a default in a truly unambiguous case, which should still cover the most common case of the user just writing to a new/empty file.